src/mark.js

```diff
@@ -385,6 +386,79 @@ class Mark { // eslint-disable-line no-unused-vars
     }
 
     /**
+     * Check if a value is a number
+     * @param {number|string} value - the value to check;
+     * numeric strings allowed
+     * @return {boolean}
+     * @access protected
+     */
+    isNumeric(value) {
+        // http://stackoverflow.com/a/16655847/145346
+        // eslint-disable-next-line eqeqeq
+        return Number(parseFloat(value)) == value;
+    }
+
+    /**
+     * @typedef Mark~setOfRanges
+     * @type {object[]}
+     * @property {number} start - The start position within the composite value
+     * @property {number} end - The end position within the composite value
+     * @property {number} len - The length of the string to mark within the
+     * composite value. If set, this value will override the end value
+     */
+    /**
+     * Returns a processed list of integer offset indexes that do not overlap
+     * each other, and remove any string values or additional elements
+     * @param {Mark~setOfRanges} array - unprocessed raw array
+     * @return {Mark~setOfRanges} - processed array with any invalid entries
+     * removed
+     * @throws Will throw an error if an array of objects is not passed
+     * @access protected
+     */
+    checkRanges(array) {
+        // start, end or length indexes are included in an array of objects
+        // [{start: 0, end: 1}, {start: 4, len: 5}]
+        // quick validity check of the first entry only
+        if (
+            !Array.isArray(array) ||
+            Object.prototype.toString.call( array[0] ) !== "[object Object]"
+        ) {
+            throw new Error("markRange() will only accept an array of objects");
+        }
```

Comment:
> If this check is removed...

Ok, I'll take care of it.

> I'd vote for one syntax: Either `len` or `end`.

Initially I had it set to `start` and `end` because it made sense to me, but there were two requests to include a length. Maybe I can reword it to say, use `start` and `end` or `start` and `len`. Or do you still want to completely remove one of them?